src/loaders/earthEngineLoader.js

```diff
@@ -79,13 +79,12 @@ const earthEngineLoader = async config => {
 
     layer.legend = {
         title: name,
-        period: period ? period.name : null,
+        period,
         groups,
         unit,
         description,
         source,
         sourceUrl,
-        ...layer.legend,
```

Comment:
Is the removal of this related to the fix?

Reply:
Yes, it should not be there, and was overwriting the new period name.